pkg/generate/dockerfile/finder.go

```diff
@@ -42,26 +42,26 @@ func NewFinder() Finder {
 	return &finder{fsWalk: filepath.Walk}
 }
 
-// Find returns a list of of found Dockerfile(s) in the given directory
+// Find returns the relative paths of Dockerfiles in the given directory dir.
 func (f *finder) Find(dir string) ([]string, error) {
 	dockerfiles := []string{}
 	err := f.fsWalk(dir, func(path string, info os.FileInfo, err error) error {
-		if info == nil {
-			return nil
+		if err != nil {
+			return err
 		}
-		// Skip hidden directories
+		// Skip hidden directories.
 		if info.IsDir() && strings.HasPrefix(info.Name(), ".") {
 			return filepath.SkipDir
 		}
-
-		// Add files named Dockerfile
-		if info.Name() == "Dockerfile" && err == nil {
+		// Add relative path to Dockerfile.
+		if info.Mode().IsRegular() && info.Name() == "Dockerfile" {
```

Comment:
If we supplying the "dir" can't we just supply the full path to Dockerfile? I can imagine the file has a different name for example. Also we can default to "./Dockerfile".